[ci_runner] Print disk stats even if there's a bazel error

maggie-lou opened the pull request.

bduffany reviewed (approved):
This LGTM, with a caveat that if most of the disk usage is coming from the `.git` directory then the `runGitMaintenance` setup will clean it up, and the disk usage fraction will go below the `highDiskUsageThreshold` and we won't wind up printing out the disk stats.

I think if we printed something like `git maintenance: cleaned <x> MB` after we call `runGitMaintenance`, that would give us the missing info there - wdyt?

maggie-lou commented:
> I think if we printed something like `git maintenance: cleaned <x> MB` after we call `runGitMaintenance`, that would give us the missing info there - also seems useful to understand whether that git maintenance step is working well in general. wdyt?

Good idea. Updated

sluongng commented:
Semi-related to the convo above https://github.com/buildbuddy-io/buildbuddy/pull/8233

goniz commented:
🙏🏻
